src/ng/rootScope.js

```diff
@@ -1350,6 +1311,24 @@ function $RootScopeProvider() {
 
     return $rootScope;
 
+    function invokeListeners(scope, event, listenerArgs, name) {
+      var listeners = scope.$$listeners[name];
+      if (listeners) {
+        if (listeners.$$index !== -1) {
+          throw $rootScopeMinErr('inevt', '{0} already $emit/$broadcast-ing', name);
+        }
+        event.currentScope = scope;
+        for (listeners.$$index = 0; listeners.$$index < listeners.length; listeners.$$index++) {
+          try {
+            //allow all listeners attached to the current scope to run
+            listeners[listeners.$$index].apply(null, listenerArgs);
+          } catch (e) {
+            $exceptionHandler(e);
```

Comment:
If `$exceptionHandler` throws, `listeners.$$index` will be never reset and the event will be never allowed to be emitted/broadcasted any more. I wonder if that would break some usecase unnecessarily.

Reply:
Yeah I'd like to fix this somehow...

Reply:
Could you just wrap the loop in a `try...finally` and put the `listerners.$$index = -1` statement in the finally block?

Reply:
I added the `try...finally` and added a test that fails without that fix.